### PR TITLE
Add NULLS FIRST / NULLS LAST qualifiers for ordering

### PR DESCRIPTION
**Pull request by matthewbauer**
Before submitting your PR, check that you've:

- [ ] Bumped the version number.
- [ ] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html).
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock.
- [ ] Ran `stylish-haskell` and otherwise adhered to the [style guide](https://github.com/bitemyapp/esqueleto/blob/master/style-guide.yaml).

After submitting your PR:

- [ ] Update the Changelog.md file with a link to your PR.
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts).

### CODE REVIEW

File: src/Database/Esqueleto/PostgreSQL.hs [hunk 1]
```diff
@@ -434,3 +438,21 @@ values exprs = Ex.From $ do
...
             )
+
+-- | Ascending order of this field or SqlExpression with nulls coming first.
+ascNullsFirst :: PersistField a => SqlExpr (Value a) -> SqlExpr OrderBy
+ascNullsFirst = orderByExpr " ASC NULLS FIRST"
+
+-- | Ascending order of this field or SqlExpression with nulls coming last.
+-- Note that this is the same as normal ascending ordering in Postgres, but it has been included for completeness.
+ascNullsLast :: PersistField a => SqlExpr (Value a) -> SqlExpr OrderBy
+ascNullsLast = orderByExpr " ASC NULLS FIRST"
```
Comment (belevy) [inline src/Database/Esqueleto/PostgreSQL.hs:449]: should be NULLS LAST.

File: src/Database/Esqueleto/PostgreSQL.hs [hunk 2]
```diff
@@ -434,3 +438,21 @@ values exprs = Ex.From $ do
...
+
+-- | Descending order of this field or SqlExpression with nulls coming first.
+-- Note that this is the same as normal ascending ordering in Postgres, but it has been included for completeness.
+ascNullsFirst :: PersistField a => SqlExpr (Value a) -> SqlExpr OrderBy
+ascNullsFirst = orderByExpr " ASC NULLS FIRST"
```
Comment (belevy) [inline src/Database/Esqueleto/PostgreSQL.hs:454]: this is `ascNullsFirst` repeated rather than `descNullsFirst`

**Comment by belevy** [comment]
Please add tests to verify this behavior works as intended.

File: test/PostgreSQL/Test.hs [hunk 3]
```diff
@@ -1339,6 +1339,54 @@ testSubselectAliasingBehavior = do
...
+testPostgresqlNullsOrdering :: SpecDb
+testPostgresqlNullsOrdering = do
+  describe "Postgresql NULLS orderings work" $ do
+      itDb "ASC NULLS FIRST works" $ do
+        p1e <- insert' p1
+        p2e <- insert' p2 -- p2 has a null age
+        p3e <- insert' p3
+        p4e <- insert' p4
+        ret <- select $
+               from $ \p -> do
```
Comment (belevy) [inline test/PostgreSQL/Test.hs:1351]: Maybe switch to using the new syntax?

**Review by belevy — APPROVED** [review]
LGTM, go ahead and add the changelog and bump the version

**Comment by sebastiant** [comment]
Would be great to see this merged!

**Pull request merged**